### PR TITLE
feature/ICL-57 update ethers to v5

### PR DESCRIPTION
**Pull request by JGiter**
- upgrade `ehters` to be compatible with updated iam-client-lib
- since `typechain` has plugin design it doesn't seems possible to generate type definition for both v4 and v5 `ethers`, `ehers-v4` folder is removed
- instead of exporting contract types will be exported build contracts. It allows clients to build contract types the way they need

### CODE REVIEW

**Review by ahmedolaibrahim — APPROVED** [review]

File: src/DomainHierarchy.ts [hunk 1]
```diff
@@ -197,7 +198,8 @@ export class DomainHierarchy {
     const logs = await provider.getLogs(filter);
     const rawLogs = logs.map(log => {
       const parsedLog = contractInterface.parseLog(log);
-      return parsedLog.values;
+      /** ethers_v5 Interface.parseLog incorrectly parses log, so have to use lowlevel alternative */
```
Comment (jrhender) [inline src/DomainHierarchy.ts:201]: Interesting, good comment 👍 . Were you able to find any open issues in ethers repo about this? Or how did you figure this out?

**Review by jrhender — APPROVED** [review]
Looks good, just one question on `Interface.parseLog` issue

File: src/DomainHierarchy.ts (hunk 1 above)
Comment (JGiter) [reply src/DomainHierarchy.ts:201]: I haven't found such issue. Do you think I may create one? I just found `decodeEventLog` in the documentation, where `parseLog` mentioned as more convenient alternative

Comment (jrhender) [reply src/DomainHierarchy.ts:201]: Sure, if you have time to create an issue in the ethers repo, I think that would be a good idea. Not a high priority though.

Comment (JGiter) [reply src/DomainHierarchy.ts:201]: Not sure if this is a bug, so I've started discussion https://github.com/ethers-io/ethers.js/discussions/1842

File: package.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
     "dist"
   ],
   "scripts": {
-    "prebuild": "rimraf dist && rimraf ethers-v4 && rimraf ethers-v4-for-tests && patch-package",
-    "build": "npm run compile:contracts && npm run build:typechain && npm run compile:ts && copyfiles \"ethers-v4/**/*.d.ts\" dist",
+    "prebuild": "rimraf dist ethers && patch-package",
+    "build": "npm run compile:contracts && npm run build:typechain && npm run compile:ts && copyfiles build/contracts/**/*.json -u 1 dist",
```
Comment (jrhender) [inline package.json:17]: @JGiter I know this is merged already, but what was the rationale for the `-u 1`? Is seems this slices off a path off the bottom of the path, but it's to clear to me why we want to do that. https://github.com/calvinmetcalf/copyfiles#command-line

Comment (jrhender) [reply package.json:17]: @JGiter ah, I see, thanks. Hmm, maybe what we should do is add copying the types from the `ethers` dir like `iam-client-lib` is doing: https://github.com/energywebfoundation/iam-client-lib/blob/2586770e5a0f1c6986392b718f181f370bd5a8a4/package.json#L14

**Pull request merged**